src/advanced/blockchain-explorer.yaml

```diff
@@ -0,0 +1,528 @@
+                      prev_hash: "87d3158f91ce3b3f8c8fab51ab15f0ddcb92a322b962c5a2b34299456c28f452"
+                      proposer_id: 3
+                      schema_version: 0
+                      state_hash: "8ad43ece286a45b6269183cc3fab215066c31054b06c9ef391697b88c03bb63c"
+                      tx_count: 1000
+                      tx_hash: "ad72304da7ded4039c54523c6e8372571df8851707e608c60f7a31d1e77515ad"
+                    - height: 10
+                      prev_hash: "f9b4f236d5ef96fe9b06e77ff5e6f9c9c6352e2822846e98d427147ee89d0aee"
+                      proposer_id: 3
+                      schema_version: 0
+                      state_hash: "8ad43ece286a45b6269183cc3fab215066c31054b06c9ef391697b88c03bb63c"
+                      tx_count: 1000
+                      tx_hash: "687d757b742a28110c40d9246586767a49b4c2521732b9ac23686481168d7e6c"
+                    - height: 6
+                      prev_hash: "f9ecccb5be51363744cffbb0b8241c2541c3aa7e53f1b9cca01d49d4991ae693"
+                      proposer_id: 3
+                      schema_version: 0
+                      state_hash: "8ad43ece286a45b6269183cc3fab215066c31054b06c9ef391697b88c03bb63c"
+                      tx_count: 1000
+                      tx_hash: "577b4d67d045d8aa08e1b63a08fe77f09ac2ae8b2f46a6c3294e4ef4efaceb6c"
+                    - height: 2
+                      prev_hash: "d8d583444b823db312be85dc5dfa6d41b658c2bfe8caf97cb4b7372f2154ad4b"
+                      proposer_id: 3
+                      schema_version: 0
+                      state_hash: "8ad43ece286a45b6269183cc3fab215066c31054b06c9ef391697b88c03bb63c"
+                      tx_count: 1000
+                      tx_hash: "94f251c0350c95024f46d26cbe0f9d2ea309e2817da4bab575fc4c571140291f"
+                  range:
+                    start: 2
+                    end: 101
+
+components:
+  schemas:
+    Hash:
+      type: string
+      example: "94f251c0350c95024f46d26cbe0f9d2ea309e2817da4bab575fc4c571140291f"
+      description: >
+        Hexadecimal string representing a hash (64 hex digits).
+    Signature:
+      type: string
+      example: "2eab829b3fc123025df6adac3f06bbafcd7882cee7601ad7791e5cc1171349c9f107b229543ee89cff2c323aafef228e850da36c4578c6c593fbb085a079d60e"
+      description: >
+        Hexadecimal string representing a signature (128 hex digits).
+    Time:
+      type: string
+      example: "2018-05-17T10:45:56.057753Z"
+      description: >
+        A string which combines date and time in UTC
+        as per [ISO 8601](https://en.wikipedia.org/wiki/ISO_8601).
+    TransactionType:
+      type: string
+      description: >
+        Transaction type, can be:
+
+        - **committed:** committed transaction (in blockchain)
+
+        - **in-pool:** uncommitted transaction (in the pool of unconfirmed transactions)
+
+        - **unknown:** unknown transaction
+      enum:
+      - committed
+      - in-pool
+      - unknown
+    TransactionStatus:
```

Comment:
Maybe, it could make more sense to represent the status as `oneOf` with a discriminator?

Reply:
I'm not sure. It does make sense from the rigorous point of view, but I kinda doubt that we'll get any merit from it.

`oneOf` does not seem to allow to specify which discriminator values choose which variants. It is a way to join two very different schemas into one. It is used for `KnownTransaction` here (and could have been used for `UnknownTransaction` as well), but that's because committed and uncommitted transactions have very different representations.

In this case we still have a single `TransactionStatus` type with a single purpose. The most common way of handling it should be check the `type` field for `"success"` and if it's not it then propagate information from `code` and `description`. I think we can leave it as a single type with some meaningful default values for `code` and `description` and don't bother with accurate discriminated union representation.

Though, here's how it's represented in the code:

```rust
enum TxStatus<'a> {
    Success,
    Panic { description: &'a str },
    Error { code: u8, description: &'a str },
}
```

so I guess we should translate that directly into `oneOf`. Maybe some day we'll have OpenAPI docs generated from the code. If I were the generator I would translate that `enum` as `oneOf`.